### DISCUSSION TITLE
[Bug]: Apparent error in internal calculation

### DISCUSSION THREAD

**Discussion by Psicrow925** [wlcrs/huawei_solar#648] [General]
### Describe the issue

it seems that either the sensor or the calculation for the charging power of the battery, or the calculation of the solar production is faulty.

the battery charge is higher than the solar production.
this means that all further calculations for the house consumption are in vain.

### Bescribe your Huawei Solar Setup

Inverter Type: SUN2000-10KTL-M1SUN2000-12
Inverter Firmware version:
SDongle present: yes
Power meter present: three phase
Battery: LUNA2000 10 kWh
Battery Firmware version:

### How do you connect to the inverter?

Dongl with WLAN

### Upload your Diagnostics File

Drag & Drop your Diagnostics File here.
[config_entry-huawei_solar-5ee521040e73cf236a33c56ea0e52dbe.json](https://github.com/wlcrs/huawei_solar/files/14549846/config_entry-huawei_solar-5ee521040e73cf236a33c56ea0e52dbe.json)

### Upload your relevant debug logs

```text
no logs
```

### Please confirm the following:

- [X] I'm running the latest release of Home Assistant.
- [X] I'm running the latest release of this integration.
- [X] I did not find an existing issue describing this problem.
- [X] I did upload the diagnostics-file that I could retrieve from  the 'Devices & Services Page'
- [ ] I did upload the relevant debug logs (via 'Enable Debug Logging'-feature or by manually configuring HA logging)

**Comment by Psicrow925** [comment]
![Screenshot 2024-03-10 113955](https://github.com/wlcrs/huawei_solar/assets/63744267/42f7c6ed-7e4b-4b58-b07d-025ccfad9b27)
![Screenshot 2024-03-10 114026](https://github.com/wlcrs/huawei_solar/assets/63744267/d0983cd9-ee18-4bad-80d0-c70585e9b16b)
![Screenshot 2024-03-10 114041](https://github.com/wlcrs/huawei_solar/assets/63744267/41293009-7803-4343-9f33-eb596959fec7)
![Screenshot 2024-03-10 115642](https://github.com/wlcrs/huawei_solar/assets/63744267/eb62d9a9-2018-4c02-9fb8-6d98bde7aead)

**Comment by Psicrow925** [comment]
![Screenshot_20240310_122117_Enpal](https://github.com/wlcrs/huawei_solar/assets/63744267/fdddd570-9c6d-4ccb-bed0-9fc26584c984)

**Comment by wlcrs** [comment]
This integration does not do any interpretation of the values provided by the inverter.

This is thus not a bug of this integration, but a problem with how these values should be interpreted. I'm therefore converting this bugreport into a discussion.
